polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala

```diff
@@ -109,12 +112,12 @@ final class RuntimeSymbolTable(
     // don't need to hash everything to determine hash code; name collisions are less common than hash comparisons
     override def hashCode(): Int = name.hashCode()
 
-    def toResultValue: ResultValue = ResultValue(kernelContext, name, scalaTypeHolder, value, sourceCellId)
+    def toResultValue: ResultValue = ResultValue(kernelContext, name, reprs, scalaTypeHolder, value, sourceCellId)
   }
 
   object RuntimeValue {
     def apply(name: String, value: Any, source: Option[LanguageInterpreter[IO]], sourceCell: CellID): RuntimeValue = RuntimeValue(
-      name, value, kernelContext.inferType(value), source, sourceCell
+      name, value, kernelContext.reprsOf(value, kernelContext.inferType(value)), kernelContext.inferType(value), source, sourceCell
```

Comment:
Can we `inferType` only once here? It's actually not that cheap to do :(

Reply:
Thanks for the catch, done!